src/pyEQL/pourbaix/pourbaix_diagram.py

```diff
@@ -437,6 +437,7 @@ def __init__(
         self,
         entries: list[PourbaixEntry] | list[MultiEntry],
         comp_dict: dict[str, float] | None = None,
+        multi_comp_dict: list[dict[str, float]] | None = None,
```

Comment:
You've got the right idea here, but it should be possible to make these changes without adding an additional kwarg, e.g.

```
comp_dict: list[dict[str, float]] | dict[str, float] | None = None,
```

The key is to place a statement in the code that converts `comp_dict` into a list, no matter what (see next comment), and then make sure anything that depends on `comp_dict` expects a list. If there's only a single element in the list, you will recover the current behavior.